site/content/docs/main/csi-snapshot-data-movement.md

```diff
@@ -0,0 +1,390 @@
+
+If you are using a customized data mover, follow the data mover's instruction for additional troubleshooting methods.  
+
+
+## How backup and restore work
+
+CSI snapshot data movement is a combination of CSI snapshot and data movement, which is jointly executed by Velero server, CSI plugin and the data mover. 
+This section lists some general concept of how CSI snapshot data movement backup and restore work. For the detailed mechanisms and workflows, you can check the [Volume Snapshot Data Movement design][1].  
+
+### Custom resource and controllers
+
+Velero has three custom resource definitions and associated controllers:
+
+- `DataUpload` - represents a data upload of a volume snapshot. The CSI plugin creates one `DataUpload` per CSI snapshot. Data movers need to handle these CRs to finish the data upload process.  
+Velero built-in data mover runs a controller for this resource on each node (in node-agent daemonset). Controllers from different nodes may handle one CR in different phases, but finally the data transfer is done by one single controller which will call uploaders from the backend.  
+
+- `DataDownload` - represents a data download of a volume snapshot.  The CSI plugin creates one `DataDownload` per volume to be restored. Data movers need to handle these CRs to finish the data upload process.  
+Velero built-in data mover runs a controller for this resource on each node (in node-agent daemonset). Controllers from different nodes may handle one CR in different phases, but finally the data transfer is done by one single controller which will call uploaders from the backend. 
+
+- `BackupRepository` - represents/manages the lifecycle of Velero's backup repositories. Velero creates a backup repository per namespace when the first CSI snapshot backup/restore for a namespace is requested. You can see information about your Velero's backup repositories by running `velero repo get`.  
+This CR is used by Velero built-in data movers, customized data movers may or may not use it.  
+
+For other resources or controllers involved by customized data movers, check the data mover's instructions.  
+
+### Backup
+
+Velero backs up resources for CSI snapshot data movement backup in the same way as other backup types. When it encounters a PVC, particular logics will be conducted:  
+
+- When it finds a PVC object, Velero calls CSI plugin though a Backup Item Action.  
+- CSI plugin first takes a CSI snapshot to the PVC by creating the `VolumeSnapshot` and  `VolumeSnapshotContent`.  
+- CSI plugin checks if a data movement is required, if so it creates a `DataUpload` CR and then returns to Velero backup.  
+- Velero now is able to back up other resources, including other PVC objects.  
+- Velero backup periodically query the data movement status from CSI plugin, CSI plugin turns to check the phase of the `DataUpload` CRs.  
```

Comment:
maybe also mention about the `itemOperationSyncFrequency` ?

Reply:
Done, thanks.